pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,629 @@
+    // it as the cgroup parent for the PodSandbox. For some runtimes, this is
+    // sufficent. For others, e.g., hypervisor-based runtimes, explicit
+    // resource limits for the sandbox are needed at creation time.
+    optional PodSandboxResources resources  = 6;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string, string> labels = 7;
+    // Annotations is an unstructured key value map that may be set by external
+    // tools to store and retrieve arbitrary metadata.
+    map<string, string> annotations = 8;
+    // Linux contains configurations specific to Linux hosts.
+    optional LinuxPodSandboxConfig linux = 9;
+}
+
+// DNSOption specifies the DNS servers and search domains.
+message DNSOption {
+    // Servers is a list of DNS servers of the cluster.
+    repeated string servers = 1;
+    // Searches is a list of DNS search domains of the cluster.
+    repeated string searches = 2;
+}
+
+// PortMapping specifies the port mapping configurations of sandbox
+message PortMapping {
+    // Name of the port mapping
+    optional string name = 1;
+    // Protocol of the port mapping.
+    // Valid candidates are: TCP, UDP
+    required string protocol = 2;
+    // The port number within the container.
+    required int32 container_port = 3;
+    // The port number on the host.
+    optional int32 host_port = 4;
+    // The host IP.
+    optional string host_ip = 5;
+}
+
+// Mount specifies the volume mount for the sandbox
+message Mount {
+    // Name of the volume mount.
+    required string name = 1;
+    // Path of the mount within the container.
+    required string container_path = 2;
+    // Path of the mount on the host.
+    required string host_path = 3;
+    // Whether the mount is read-only.
+    required bool readonly = 4;
+    // Whether the mount needs SELinux relabeling
+    optional bool selinux_relabel = 5;
+}
+
+// ResourceRequirements containers a set of resources
+// Valid resources are:
+// - cpu, in cores. (500m = .5 cores)
+// - memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
+message ResourceRequirements {
+    // Limits describes the maximum amount of compute resources allowed.
+    optional double limits = 1;
+    // Requests describes the minimum amount of compute resources required.
+    // If Request is omitted for a container, it defaults to Limits if that is explicitly specified,
+    // otherwise to an implementation-defined value
+    optional double requests = 2;
+}
+
+// PodSandboxResources contains the CPU/memory resource requirements.
+message PodSandboxResources {
+    // CPU resource requirement.
+    optional ResourceRequirements cpu = 1;
+    // Memory resource requirement.
+    optional ResourceRequirements memory = 2;
+}
+
+// NamespaceOption provides options for Linux namespaces.
+message NamespaceOption {
+    // HostNetwork uses the host's network namespace.
```

Comment:
Hm...ok. If the purpose of using `HostNetwork` instead of `host_network` in the comment is to make the comment compatible with the generated gocode, I'd suggest not mentioning the name at all. Otherwise, it'd look weird in one of the files. E.g., just `// If set, use the host's network namespace`